image/image_linux.go

```diff
@@ -392,6 +409,23 @@ func (s *imageSeeder) start(db *asserts.Database, optSnaps []*seedwriter.Options
 	return s.w.Start(db, newFetcher)
 }
 
+func (s *imageSeeder) validateSnapArchs(snaps []*seedwriter.SeedSnap) error {
```

Comment:
(super nitpick^2) and probably personal preference. Splitting into a helper might make it slightly easier to read:
```go
func (s *imageSeeder) validateSnapArch(sn *seedwriter.SeedSnap) bool {
	for _, a := range sn.Info.Architectures {
		if a == "all" || a == s.architecture {
			return true
		}
	}
	return false
}

func (s *imageSeeder) validateSnapArchs(snaps []*seedwriter.SeedSnap) error {
	for _, sn := range snaps {
		if !s.validateSnapArch(sn) {
			return fmt.Errorf("snap %q supported architectures (%s) are incompatible with the model architecture (%s)",
				sn.Info.SnapName(), strings.Join(sn.Info.Architectures, ", "), s.architecture)
		}
	}
	return nil
}
```

Reply:
love this suggestion!